src/js/elements/menu.js

```diff
@@ -1,174 +1,176 @@
-            let hasChildren = checkIfSubCategoryHasChildren(subcategory, detail);
-
-            if (hasChildren) {
-                let count = subindicatorsInSubCategory(detail);
-                if (count > 0) {
-                    addIndicators(h2Wrapper, category, subcategory, detail.indicators);
-                }
-            }
+  }
+
+
+  addSubcategories(wrapper, category, subcategories) {
+    var newCategory = categoryTemplate.cloneNode(true)
+    $(newCategory).removeClass(hideondeployClsName);
+    $(".data-category__h1_title div", newCategory).text(category);
+    $('.' + loadingClsName).addClass('hidden');
+    $('.' + noDataWrapperClsName).addClass('hidden');
+    parentContainer.append(newCategory);
+    var h2Wrapper = $(".data-category__h1_wrapper", newCategory);
+    $(".data-category__h2", h2Wrapper).remove();
+
+    for (const [subcategory, detail] of Object.entries(subcategories)) {
+      let hasChildren = checkIfSubCategoryHasChildren(subcategory, detail);
+
+      if (hasChildren) {
+        let count = subindicatorsInSubCategory(detail);
+        if (count > 0) {
+          this.addIndicators(h2Wrapper, category, subcategory, detail.indicators);
         }
+      }
     }
+  }
 
-    function setActive(el) {
-        resetActive();
-        $(this).addClass("menu__link_h4--active")
-    }
+  setActive(el) {
+    this.resetActive();
+    $(this).addClass("menu__link_h4--active")
+  }
 
-    function resetActive() {
-        $(".menu__link_h4--active", parentContainer).removeClass("menu__link_h4--active");
-    }
+  resetActive() {
+    $(".menu__link_h4--active", parentContainer).removeClass("menu__link_h4--active");
+  }
 
+
+  main() {
     $(".data-menu__category").remove();
     let hasNoItems = true;
     let hiddenClass = hideondeployClsName;
     hiddenClass = 'hidden';
     $(parentContainer).find('.data-category').remove();
+    for (const [category, detail] of Object.entries(this._data)) {
+      let hasChildren = checkIfCategoryHasChildren(category, detail)
 
-
-    for (const [category, detail] of Object.entries(data)) {
-        let hasChildren = checkIfCategoryHasChildren(category, detail)
-
-        if (hasChildren) {
-            if (!$('.' + noDataWrapperClsName).hasClass(hiddenClass)) {
-                $('.' + noDataWrapperClsName).addClass(hiddenClass);
-            }
-            hasNoItems = false;
-
-            addSubcategories(parentContainer, category, detail.subcategories)
+      if (hasChildren) {
+        if (!$('.' + noDataWrapperClsName).hasClass(hiddenClass)) {
+          $('.' + noDataWrapperClsName).addClass(hiddenClass);
         }
+        hasNoItems = false;
+
+        this.addSubcategories(parentContainer, category, detail.subcategories)
+      }
     }
 
     if (hasNoItems) {
-        dataMapperMenu.showNoData()
+      this._dataMapperMenu.showNoData()
     }
+  }
+
+
 }
 
 /**
 * This class is a stub for a menu component
 */
 export class DataMapperMenu extends Component {
```

Comment:
I think my idea with this class was to act as the Component itself, i.e. your LoadMenu component